CONTRIBUTING.rst

```diff
@@ -0,0 +1,678 @@
+-  `Migrating Multiple Repositories to GitHub Apps Guide <https://docs.travis-ci.com/user/travis-migrate-to-apps-gem-guide/>`__.
+
+Changing the Metadata Database
+==============================
+
+When developing features, you may need to persist information to the metadata
+database. Airflow has `Alembic <https://bitbucket.org/zzzeek/alembic>`__. built-in
+module to handle all schema changes. Alembic must be installed on your
+development machine before continuing with migration.
+
+
+.. code-block:: bash
+
+    # starting at the root of the project
+    $ pwd
+    ~/airflow
+    # change to the airflow directory
+    $ cd airflow
+    $ alembic revision -m "add new field to db"
+       Generating
+    ~/airflow/airflow/migrations/versions/12341123_add_new_field_to_db.py
+
+
+Setting up the node / npm Javascript Environment
+================================================
+
+`airflow/www/` contains all npm-managed, front-end assets. Flask-Appbuilder
+itself comes bundled with jQuery and bootstrap. While they may be phased out
+over time, these packages are currently not managed with npm.
+
+Make sure you are using recent versions of node and npm. No problems have been
+found with node\>=8.11.3 and npm\>=6.1.3.
+
+Installing npm and its packages
+-------------------------------
+
+Make sure npm is available in your environment.
+
+To install it on macOS:
+
+1.  Run the following commands (taken from [this
+    source](https://gist.github.com/DanHerbert/9520689)):
+
+.. code-block:: bash
+
+    brew install node --without-npm
+    echo prefix=~/.npm-packages >> ~/.npmrc
+    curl -L https://www.npmjs.com/install.sh | sh
+
+
+2.  Add ``~/.npm-packages/bin`` to your ``PATH`` so that commands you install
+    globally are usable.
+
+3.  Set up your ``.bashrc`` file and then ``source ~/.bashrc`` to reflect the
+    change.
+
+    For example:
+
+.. code-block:: bash
+
+    export PATH="$HOME/.npm-packages/bin:$PATH"
+
+
+    You can also follow  `general npm installation
+     instructions <https://docs.npmjs.com/downloading-and-installing-node-js-and-npm>`__..
+
+4.  Install third party libraries defined in ``package.json`` by running the
+    following commands within the ``airflow/www/`` directory:
+
+
+.. code-block:: bash
+
+    # from the root of the repository, move to where our JS package.json lives
+    cd airflow/www/
+    # run npm install to fetch all the dependencies
+    npm install
+
+
+These commands install the libraries in a new ``node_modules/`` folder within
+``www/``.
+
+Should you add or upgrade an npm package, which involves changing
+``package.json``, you'll need to re-run ``npm install`` and push the newly generated
+``package-lock.json`` file so that we get a reproducible build.
+
+
+Generating Bundled Files with npm
+---------------------------------
+
+To parse and generate bundled files for Airflow, run either of the following
```

Comment:
Elena: don't we want to explain what 'bundled files' mean?
Jarek: I don't think it's needed. Bundling NPM packages is very well known term for Javascript developers.